joint_trajectory_controller/include/joint_trajectory_controller/init_joint_trajectory.h

```diff
@@ -66,9 +66,6 @@ inline std::vector<unsigned int> permutation(const T& t1, const T& t2)
 {
   typedef unsigned int SizeType;
 
-  // Arguments must have the same size
-  if (t1.size() != t2.size()) {return std::vector<SizeType>();}
```

Comment:
I have also removed the permutation variable inside a state and segment as they are now a set for a specific joint.